packages/apps/plugins/plugin-explorer/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useResizeDetector } from 'react-resize-detector';
+
+import { type Space } from '@dxos/client/echo';
+import { createSvgContext, SVG, SVGContextProvider } from '@dxos/gem-core';
+
+import { HierarchicalEdgeBundling, RadialTree, TidyTree } from './layout';
+import { mapGraphToTreeData, type TreeNode } from './types';
+import { SpaceGraphModel } from '../Graph';
+
+// TODO(burdon): Create dge bundling graph using d3.hierarchy.
+// https://observablehq.com/@d3/hierarchical-edge-bundling?intent=fork
+
+type Renderer = (svg: SVGSVGElement, data: any, options: any) => void;
+
+export type LayoutType = 'tidy' | 'radial' | 'edge';
+
+// TODO(burdon): Normalize API and styling.
+
+export type TreeLayoutSlots = {
+  node?: string;
+  path?: string;
+  text?: string;
+};
+
+export type TreeOptions = {
+  label: (d: any) => string;
+
+  slots?: TreeLayoutSlots;
+  radius?: number;
+
+  width: number;
+  height: number;
+  margin?: number;
+
+  padding?: number;
+  // Radius of nodes.
+  r?: number;
+};
+
+export const defaultTreeLayoutSlots: TreeLayoutSlots = {
+  node: 'fill-blue-600',
+  path: 'fill-none stroke-blue-400 stroke-[0.5px]',
+  text: 'stroke-[0.5px] stroke-neutral-700 text-xs', // TODO(burdon): Create box instead of halo.
+};
```

Comment:
These should be organized the same way as all other styles are organized. They should go in a `theme.ts` or similar, and export strings/`ComponentFragment`s/`ComponentFunction`s the same as other theme files.

Reply:
For an example in an intermediate component like this, see https://github.com/dxos/dxos/blob/main/packages/ui/react-ui-table/src/theme.ts

Reply:
Thanks for the reference -- that's helpful.
I had meant to bring up for a while though a slight concern I have about "polluting" (too strong a term) the global namespace with style fragments (esp. given that there are so many of them). Would like to discuss.

Reply:
Sure — currently we have two precedents:
- low-level components must set styles in theme objects (which happen to resemble the old slot objects) in `react-ui-theme`
- intermediate components are allowed to set styles directly in any `className` or `classNames` props, and some, like Table, provide theme objects at the package root, as if they might migrate directly to `react-ui-theme`.

I could see cause for defining theme objects elsewhere, especially for plugin components like this – location seems less important than that style defaults & overrides are defined and applied consistently.